src/common/main/kotlin/me/owdding/customscoreboard/config/MainConfig.kt

```diff
@@ -77,55 +112,67 @@ object MainConfig : ConfigKt("customscoreboard/config") {
         this.translation = "customscoreboard.config.enabled"
     }
 
-    private val default = listOf<ScoreboardEntry>(
-        ScoreboardEntry.TITLE,
-        ScoreboardEntry.LOBBY,
-        ScoreboardEntry.SEPARATOR,
-        ScoreboardEntry.DATE,
-        ScoreboardEntry.TIME,
-        ScoreboardEntry.ISLAND,
-        ScoreboardEntry.AREA,
-        ScoreboardEntry.PROFILE,
-        ScoreboardEntry.SEPARATOR,
-        ScoreboardEntry.PURSE,
-        ScoreboardEntry.MOTES,
-        ScoreboardEntry.BANK,
-        ScoreboardEntry.BITS,
-        ScoreboardEntry.COPPER,
-        ScoreboardEntry.GEMS,
-        ScoreboardEntry.HEAT,
-        ScoreboardEntry.COLD,
-        ScoreboardEntry.NORTH_STARS,
-        ScoreboardEntry.SOULFLOW,
-        ScoreboardEntry.SEPARATOR,
-        ScoreboardEntry.OBJECTIVE,
-        ScoreboardEntry.SLAYER,
-        ScoreboardEntry.QUIVER,
-        ScoreboardEntry.EVENTS,
-        ScoreboardEntry.POWDER,
-        ScoreboardEntry.MAYOR,
-        ScoreboardEntry.PARTY,
-        ScoreboardEntry.PET,
-        ScoreboardEntry.FOOTER,
-    )
+    private val default = listOf(
+        ElementTitle,
+        ElementLobby,
+        ElementSeparator,
+        ElementDate,
+        ElementTime,
+        ElementIsland,
+        ElementArea,
+        ElementProfile,
+        ElementSeparator,
+        ElementPurse,
+        ElementMotes,
+        ElementBank,
+        ElementBits,
+        ElementCopper,
+        ElementGems,
+        ElementHeat,
+        ElementCold,
+        ElementNorthStars,
+        ElementSoulflow,
+        ElementSeparator,
+        ElementObjective,
+        ElementSlayer,
+        ElementQuiver,
+        ElementEvents,
+        ElementPowder,
+        ElementMayor,
+        ElementParty,
+        ElementPet,
+        ElementFooter,
+    ).joinToString(",") { it.id }
 
     val appearance by observable(
-        draggable(*default.toTypedArray()) {
-            this.translation = "customscoreboard.config.appearance"
-            this.duplicatable = listOf(ScoreboardEntry.SEPARATOR).toTypedArray()
-        },
-    ) { old, new ->
+        transform(
+            string(default) {
+                this.translation = "customscoreboard.config.appearance"
+                renderer = CUSTOM_DRAGGABLE_RENDERER
+            },
```

Comment:
this should just use a string array instead, no reason to switch it from an array of strings to a comma separated string